src/Module/Util/ZipHandler.php

```diff
@@ -57,27 +58,41 @@ public function isZipable(string $object_type): bool
 
     /**
      * takes array of full paths to medias
-     * zips them and sends them
+     * zips them, adds art and m3u, and sends them
      *
      * @param string $name name of the zip file to be created
      * @param array $media_files array of full paths to medias to zip create w/ call to get_media_files
      */
     public function zip(string $name, array $media_files): void
     {
+        $art     = (string) AmpConfig::get('album_art_preferred_filename');
+        $addart  = (bool) AmpConfig::get('art_zip_add');
         $filter  = preg_replace('/[^a-zA-Z0-9. -]/', '', $name);
         $arc     = new ZipStream($filter . ".zip");
+        $pl      = '';
         $options = [
-            'comment' => $this->configContainer->get('file_zip_comment'),
+            'comment' => AmpConfig::get('file_zip_comment'),
         ];
-
+        
         foreach ($media_files as $dir => $files) {
             foreach ($files as $file) {
+                $dirname = dirname($file);
+                $artpath = dirname($file) . "/" . $art;
+                $folder  = explode("/", $dirname)[substr_count ($dirname, "/", 0)];
+                $pl      = $folder . "/" . basename($file);
+                try {
+                    $arc->addFileFromPath($folder . "/" . basename($file), $file, $options);
+                } catch (Exception $e) {
```

Comment:
Would it make sense to log errors in here? There is already a logger instance ready to use in the class

Reply:
Sure makes sense to have somthing in the log indicating what went wrong